eventdemo/source/objc/EventDemo/ChatClientManager.m

```diff
@@ -0,0 +1,195 @@
+/**
+ *  Copyright 2014-2016 CyberVision, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#import "ChatClientManager.h"
+
+#define USER_EXTERNAL_ID    @"user@email.com"
+#define USER_ACCESS_TOKEN   @"token"
+
+
+@implementation ChatClientManager
+
+
++ (ChatClientManager *)sharedManager //Singletone
+{
+    static ChatClientManager *sharedMyManager = nil;
+    @synchronized(self) {
+        if (sharedMyManager == nil)
+            sharedMyManager = [[self alloc] init];
+    }
+    return sharedMyManager;
+}
+
+- (id)init
+{
+    self = [super init];
+    if(self)
+    {
+        [self initializeKaa];
+        self.defaultRooms = @[@"Living", @"Guest"];
+        self.rooms = [NSMutableArray arrayWithArray:_defaultRooms];
+        messages = [NSMutableDictionary dictionary];
+    }
+    return self;
+}
+
+- (void)initializeKaa
+{
+    //Create a Kaa client with the Kaa default context.
+    self.kaaClient = [KaaClientFactory
+                      clientWithContext:[[DefaultKaaPlatformContext alloc] init]
+                      stateDelegate:self];
+    [self.kaaClient setProfileContainer:self];
+    
+    // Start the Kaa client and connect it to the Kaa server.
+    [self.kaaClient start];
+    
+    //Attaching new user
+    [self.kaaClient attachUserWithId:USER_EXTERNAL_ID
+                         accessToken:USER_ACCESS_TOKEN delegate:self];
+}
+
+#pragma mark - Kaa Delegates methods
+
+#pragma mark - KaaClientStateDelegate
+
+- (void)onStarted
+{
+    NSLog(@"Kaa client started");
+}
+
+#pragma mark - UserAttachDelegate
+
+- (void)onAttachResult:(UserAttachResponse *)response
+{    
+    if (response.result == SYNC_RESPONSE_RESULT_TYPE_SUCCESS)
+    {
```

Comment:
Curly braces general rule - place them on the same line before and with a newline after.